src/contexts/Balances/index.tsx

```diff
@@ -200,6 +203,17 @@ export const BalancesProvider = ({
           _bonded === null ? null : (_bonded.toHuman() as string | null);
         _account.bonded = _bonded;
 
+        // add bonded (controller) account as external account if not presently imported
+        if (_bonded) {
+          if (
```

Comment:
Could be cleaner with extracted function instead of nested ifs ?

Reply:
Definitely could be. This will be an entire sprint, formalising the contexts & introducing better state management.